src/mpb.cpp

```diff
@@ -422,7 +428,8 @@ void *fields::get_eigenmode(double frequency, direction d, const volume where, c
     // finally, send MPB the real epsilon data using the synchronized cache
     eps_data.use_cache = true;
     eps_data.icache = 0;
-    set_maxwell_dielectric(mdata, mesh_size, R, G, meep_mpb_eps, NULL, &eps_data);
+    set_maxwell_dielectric(mdata, mesh_size, R, G, NULL, meep_mpb_eps, &eps_data);
+    assert(eps_data.icache == eps_data.ncache);
```

Comment:
Looks like using `assert` here requires putting `insert <assert.h>` at the top of `mpb.cpp` similar to `near2far.cpp`:

https://github.com/NanoComp/meep/blob/4219b1d47db3d1c5ff41ef38f7296c463fcdc681/src/near2far.cpp#L24